src/lib/mina/transaction.ts

```diff
@@ -276,6 +280,40 @@ type RejectedTransaction = Pick<
   errors: string[];
 };
 
+interface TransactionPromise extends Promise<Transaction> {
+  sign(...args: Parameters<Transaction['sign']>): TransactionPromise;
+  send(): PendingTransactionPromise;
+}
```

Comment:
could you do this with types, not interfaces? like `Promise<Transaction> & ...`? I'd prefer that because we use types for everything else and we understand them well